source/docs/advanced/theme/sitemap.md

```diff
@@ -0,0 +1,172 @@
+
+- declares the following types
+
+  ```graphql
+  # src/server/modules/faq/schema.gql
+  type FaqEntry {
+    slug: String
+    question: String
+    answer: String
+  }
+
+  extend type Query {
+    allFaq: [FaqEntry]
+    faq(slug: String!): FaqEntry
+  }
+  ```
+
+- declares a loader that allows to fetch all the FAQ entries and returns an array of object containing the slug, the question and the answer for each FAQ entry.
+  ```js
+  // src/server/modules/faq/loaders.js
+  const FaqLoader = () => {
+    return {
+      allFaq: () => {
+        return fetch("https://my-faq-service.example.org/all").then(
+          response => response.data
+        );
+      }
+    };
+  };
+  ```
+
+#### Declare FaqEntry type as Sitemapable
+
+First thing first, we will need to edit our FaqEntry type to register it as Sitemapable:
+
+```diff
+# src/server/modules/faq/schema.gql
+-type FaqEntry {
++type FaqEntry implements Sitemapable {
+  slug: String
+  question: String
+  answer: String
++  "The location used to display this FaqEntry"
++  path: String
++  "The priority of crawling defined in the sitemap"
++  priority: Float
++  "Last modification date of the content of the page"
++  lastmod: String
++  "The change frequency of the page"
++  changefreq: String
++  "The list of images related to this page"
++  seoImages: [SitemapImage]
+}
+```
+
+`path`, `priority`, `seoImages`, `lastmod` and `changefreq` are fields that will be used to render the sitemap. To define their values, you can use the usual GraphQL way. In our case, we will use small resolvers:
+
+```diff
+// src/server/modules/faq/resolvers.js
+export default {
+  // ...
++  FaqEntry: {
++    path: ({ slug }) => `/faq/${slug}`,
++    priority: () => 0.5,
++    lastmod: () => new Date().toISOString(),
++    changefreq: () => "daily",
++    seoImages: () => []
++  }
+};
+```
+
+#### Add the question pages to the sitemap
+
+If we want these pages to appear in the sitemap, we will need to fetch all the questions and register them in the sitemap. Thus, in the `index.js` file where you have declared your FAQ GraphQL module (ex: `src/server/modules/faq/index.js`), you will need to register your nodes like this:
+
+```diff
+// src/server/modules/faq/index.js
+export default {
+  // ...
++  dependencies: ["Front-Commerce/Core"]
+-  contextEnhancer: () => {
++  contextEnhancer: ({ loaders }) => {
+    const Faq = FaqLoader()
+
++    loaders.Sitemap.registerNodesFetcher("Faq", () =>
++      Faq.loadAll().then(entries => {
++        return entries.map(entry => ({
++          ...entry,
++          __typename: "FaqEntry"
++        }));
++      })
++    );
+
+    return {
+        Faq: Faq
+    }
+  }
+  // ...
+};
+```
+
+The `Sitemap` loader is a loader that is available in the `Front-Commerce/Core` GraphQL module. Don't forget to add it. It will help you detect errors in case something is not working properly.
```

Comment:
I don't understand the meaning/goal of the "Don't forget to add it. It will help you detect errors in case something is not working properly." sentences.

Was is aimed at explaining why the module was added to dependencies?

Reply:
In practice it's optional to set the dependency here. It will work as long as the core module is in the correct order thanks to toposort. Luckily it's mostly the case even without declaring the dependencies here. But it can lead to errors if someday it's no longer available. At least if we add the dependency, an error message will be explicitly displayed upon build and explain what's happening to the user.